### PR TITLE
Consolidate build and deploy scripts fixes #249

### LINKED ISSUE #249: Consolidate arm build/deploy scripts
## Expected Behaviour

`./build.sh` should work for ARM/x86_64 etc
`./deploy.sh` should work for ARM/x86_64 etc

## Current Behaviour

`./build.armhf.sh` is needed along with `./build.sh` for x86_64.

A separate deployment script is also needed - this should be consolidated. See #248 for an example of a technique for this work.

### PR DESCRIPTION
**Pull request by rgee0**
Signed-off-by: rgee0 <richard@technologee.co.uk>

## Description
Previously the build and deploy shell scripts were distinct entities for each supported hardware architecture.  This change seeks to bring the separate scripts together and uses `uname -m` to switch the deploy script and build Dockerfile accordingly.

- Also added queue package to the armhf Dockerfile to provide consistency with the x86_64 & arm64 gateways.

- Added the ability to deploy `docker-compose.extended.yml` on x86_64 by passing in the argument of `extended` into `./deploy_stack.sh`

- Removed the defunct *.armhf.sh files.

## Motivation and Context
Simplifies the build and deploy activities which consequently improves the user experience.
Fixes issue #249
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
Tested that the shell scripts correctly switch to the relevant architecture type on `x86_64`, `armv7` and `aarch64`.
Tried the `extended` argument on all three architectures to ensure that it only had a material effect on x86_64. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: deploy_stack.sh [hunk 1]
```diff
@@ -5,6 +5,28 @@ if ! [ -x "$(command -v docker)" ]; then
...
+"aarch64" | "armv8l")           COMPOSEFILE="docker-compose.arm64.yml"
+                                 ;;
+
+*) echo "Sorry, your architecture ($ARCH) cannot currently be matched to a compose file."
+   exit 1
+   ;;
+esac
+
+if [ "$ARCH" = "x86_64" ] && [ "$1" = "extended" ] ; then
+COMPOSEFILE="docker-compose.extended.yml"
```
Comment (alexellis) [inline deploy_stack.sh]: Is there no extended compose file for arm yet? We should look at that because NATS has support as a multi-arch image now.

**Comment by rgee0** [comment]
@alexellis I've amended the not found messages to include a request that an issue be raised on GH.

File: deploy_stack.sh (hunk 1 above)
Comment (rgee0) [reply deploy_stack.sh]: #322 logged to look at extended compose file(s) for arm

**Comment by alexellis** [comment]
Thank you. I'll merge into master once we have someone else test. What blogs and documentation do we need to update?

**Comment by alexellis** [comment]
@rgee0 @JockDaRock @johnmccabe please can you help?

**Comment by alexellis** [comment]
@rgee0 what remains to be done here? See also merge conflicts.

**Comment by rgee0** [comment]
@alexellis suggest closing this. Too long has passed.  Things move on.

**Comment by alexellis** [comment]
OK can you summarise where we were with it and how I should go about moving forward if I'm going to be handling this now?

**Pull request closed without merging**